zerver/lib/events.py

```diff
@@ -511,6 +511,20 @@ def fetch_initial_state_data(
             state[notification] = getattr(settings_user, notification)
         state["available_notification_sounds"] = get_available_notification_sounds()
 
+    if want("user_settings"):
+        state["user_settings"] = {}
+
+        for prop in UserProfile.property_types:
+            state["user_settings"][prop] = getattr(settings_user, prop)
+        for notification in UserProfile.notification_setting_types:
+            state["user_settings"][notification] = getattr(settings_user, notification)
+
+        state["user_settings"]["emojiset_choices"] = UserProfile.emojiset_choices()
+        state["user_settings"]["timezone"] = settings_user.timezone
+        state["user_settings"][
+            "available_notification_sounds"
+        ] = get_available_notification_sounds()
```

Comment:
I am not sure if we should include `emojiset_choices` and `available_notification_sounds` as they are not settings actually and are just the possible choices for these settings.

Reply:
Good question; maybe one for `#api design`.  It kinda makes sense to send these whenever we're sending the `user_settings`, since they are the values that let one render UI for changing those values and are small, but I can see an argument for putting this sort of thing in a separate object.